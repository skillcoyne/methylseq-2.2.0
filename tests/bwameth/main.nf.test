nextflow_pipeline {

    name "bwameth"
    script "main.nf"

    test("Single End") {
        when {
            params {
                aligner = "bwameth"
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 31
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End with index") {
        when {
            params {
                aligner = "bwameth"
                // Generated by previous test
                bismark_index = "$outputDir/bwameth/reference_genome/bwameth/"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 31
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS") {
        when {
            params {
                aligner = "bwameth"
                save_reference = true
                skip_trimming = true
                rrbs = true
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 25
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS with index") {
        when {
            params {
                aligner = "bwameth"
                skip_trimming = true
                rrbs = true
                // Generated by previous test
                bismark_index = "$outputDir/bwameth/reference_genome/bwameth/"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 25
            // FIXME assert snapshot(workflow).match()
        }
    }

}
