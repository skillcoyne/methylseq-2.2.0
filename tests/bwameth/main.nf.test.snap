{
    "Single End RRBS with index": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/5dcf4cfe1c5c8be1bd4a82dd595f2288/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 34,
                    "tasksSucceeded": 31
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:31:52+0000"
    },
    "Single End with index": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/8bf7115fa8318e3ff0bbd8776628ae0a/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 43,
                    "tasksSucceeded": 40
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:31:52+0000"
    },
    "Single End": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/b87c472c91d742d60b3500039de13d5f/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 43,
                    "tasksSucceeded": 40
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:31:52+0000"
    },
    "Single End RRBS": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/1c27b6ff562ce0345db044e24275678c/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 34,
                    "tasksSucceeded": 31
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:31:52+0000"
    }
}
